service/controller/chart/v1/resource/release/desired.go

```diff
@@ -20,6 +20,16 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	}
 
 	releaseName := key.ReleaseName(cr)
+
+	if key.IsDeleted(cr) {
```

Comment:
I think this is the wrong place to check that. The decision and strategy implemented to decide how a certain change we want to apply should be done is in `new*Change`. `GetDesiredState` should not be conditional as such and only return the actual desired state. At least this was the intention of the interface so far.

Reply:
Yes I see what you mean. The problem is getting the configmap and secret will fail because they have already been deleted by app-operator.

I've moved the checks. So we no calculate as much of the desired state as we can. WDYT?

Reply:
Now you only moved the check to another place but the scope is the same. What about not returning an error when the resources we need in order to compute the desired state cannot be found? Then we propagate the incomplete desired state and can reason about the deletion where we should. I see that the CRUD interface here is maybe not really nice due to these limitations. We could think if we want to change this in the future. Anyway, let's find some solution now and have a TODO issue or something to improve here later. I leave that to you.

Reply:
I moved it because I thought it was clearer to have the conditional at the point it was needed.

But not returning a not found error if the CR is deleted is a better solution. Thanks! I'll implement that.

Not sure about a TODO issue. I'll have a think on that.  IMO this is a consequence of the design and having the app CR as a higher level of abstraction above the chart CR.

Reply:
Changes made. @xh3b4sd PTAL when you have time.